chapter02/worksheets/solution2.1.7.1_AR.scala

```diff
@@ -6,20 +6,12 @@ Find all pairs i, j within (0, 1, ..., 9) such that i + 4 ∗ j > i ∗ j.
 Hint: use flatMap and filter.
   */
 
-val a = (0 to 9).sliding(2).toList
-val b = a.map{ case x => (x.apply(0) +  4 * x.apply(1)) }
-val c = a.map{ case x => (x.apply(0) * x.apply(1)) } 
+val a = (0 to 9).flatMap(x => (0 to 9).map { y => (x, y) } )
+val b = a.map{ case x => x._1 + 4 * x._2 }
+val c = a.map{ case x => x._1 * x._2 }
 val d = b.zip(c)
-// List((4,0), (9,2), (14,6), (19,12), (24,20), (29,30), (34,42), (39,56), (44,72))
-val e = d.map{ case x => if (x._1  > x._2) x }.filter(_ != ())
+val result = d.map{ case x => if (x._1  > x._2) x }.filter(_ != ())
```

Comment:
Möchtest Du hier den `filter` erst anwenden und auch `case` mit Tuple?

```scala
d.filter { case (x, y) => ... }  .map { case (x, y) => ...}
```

Sonst bekommst du immer noch `Seq[Any]` und verlierst die Typen und die Möglichkeit, dass (in einer anderen möglichen Aufgabe) der Wert `()` ein erlaubter Wert sein könnte.